src/sdk/bucketspace_fs.js

```diff
@@ -650,6 +623,72 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
     async put_object_lock_configuration(params, object_sdk) {
         // TODO
     }
+
+    /////////////////
+    ///// UTILS /////
+    /////////////////
+
+    // TODO: move the following 3 functions - has_bucket_action_permission(), validate_fs_bucket_access(), _has_access_to_nsfs_dir()
+    // so they can be re-used
+    async has_bucket_action_permission(bucket, account, action, bucket_path = "") {
+        const account_identifier = account.name.unwrap();
+        dbg.log1('has_bucket_action_permission:', bucket.name.unwrap(), account_identifier, bucket.system_owner, bucket.owner_account);
+
+        const is_system_owner = account_identifier === bucket.system_owner.unwrap();
+
+        // If the system owner account wants to access the bucket, allow it
+        if (is_system_owner) return true;
+        const is_owner = (bucket.bucket_owner.unwrap() === account_identifier);
+        const bucket_policy = bucket.s3_policy;
+
+        if (!bucket_policy) return false;
```

Comment:
@romayalon should we return true when `is_owner` true and there is no policy for buckey?

Reply:
see lines 657-658, if bucket policy result is DENY we will return false

Reply:
@romayalon I didn't understand why it is different from:
https://github.com/noobaa/noobaa-core/blob/8ef4d07243d91c030a9501e8903413dca760d4df/src/server/common_services/auth_server.js#L683
Worth adding a comment with an explanation about this.